### PR TITLE
[#2803] ValidateSenderDomain operation with provider strategies

### LINKED ISSUE #2803: ValidateSenderDomain operation with provider strategies
## Parent

Blocked by / part of #2792
Depends on: Phase 1a (DomainEmailConfig model)

## Summary

Create the `ValidateSenderDomain` operation using the strategy pattern to support DNS-based domain verification across SES, SendGrid, and Lettermint providers.

## Context

This is **Phase 2** of #2792. It can run in parallel with Phases 1b and 3 once Phase 1a is complete. The operation follows the existing `VerifyDomain` operation pattern.

### Core Operation
- [ ] `lib/onetime/operations/validate_sender_domain.rb` — orchestrator that delegates to provider strategy
- [ ] Strategy selection based on `DomainEmailConfig#provider`
- [ ] Generate required DNS records (DKIM, SPF) for user to configure
- [ ] Query DNS to check verification status
- [ ] Update `DomainEmailConfig#verification_status` and `#verified_at`

### Provider Strategies
- [ ] `strategies/ses_validation.rb` — AWS SES identity verification (DKIM + SPF + MX)
- [ ] `strategies/sendgrid_validation.rb` — SendGrid domain authentication (DKIM + SPF)
- [ ] `strategies/lettermint_validation.rb` — Lettermint domain verification (DKIM + SPF)

### Verification Flow
1. User saves config with provider selection
2. Operation generates DNS records to display
3. User configures DNS at their registrar
4. User triggers validation check
5. Operation queries DNS and updates status

### Tests
- [ ] Unit tests for each provider strategy
- [ ] Integration tests for the orchestrator
- [ ] DNS record generation tests
- [ ] Status transition tests (pending → verified, pending → failed)

## References

- Existing domain verification: `lib/onetime/operations/verify_domain.rb`
- Provider comparison table in #2792

| Provider | Validation Method | Records |
|----------|------------------|---------|
| AWS SES | DNS (DKIM, SPF, MX) | 3 CNAME + 1 MX + 1 TXT |
| SendGrid | DNS (DKIM, SPF) | 3 CNAME + 1 TXT |
| Lettermint | DNS (DKIM, SPF) | 2 CNAME + 1 TXT |

## Acceptance Criteria

- Each provider strategy generates correct DNS records for display
- DNS verification correctly reports pass/fail per record type
- Verification status is persisted on the config model
- Strategy pattern allows adding new providers without modifying orchestrator

### PR DESCRIPTION
**Pull request by delano**
## Summary

Adds a provider-agnostic DNS validation framework for sender domain verification, implementing Phase 2 of #2792. Three provider strategies (SES, SendGrid, Lettermint) generate and verify the DNS records customers need for authenticated email sending (DKIM, SPF, MX).

The central `ValidateSenderDomain` operation orchestrates the flow: load domain, select strategy by provider type, verify DNS records via `Resolv::DNS`, and optionally persist verification status to `MailerConfig`.

### What's included

- **Strategy base class** with shared DNS lookup helpers (TXT, CNAME, MX) using Ruby stdlib `Resolv::DNS` — no external dependencies
- **Factory** routing provider type string to concrete strategy, with input normalization
- **Provider strategies** matching the record specs from #2792:
  - SES: 3 DKIM CNAME + 1 SPF TXT + 1 MX (5 records)
  - SendGrid: 2 DKIM CNAME + 1 link branding CNAME + 1 SPF TXT (4 records)
  - Lettermint: 2 DKIM CNAME + 1 SPF TXT (3 records)
- **Orchestrator operation** following the `VerifyDomain` pattern — immutable `Data.define` Result, optional persistence, error wrapping (never raises from `#call`)
- **SPF-aware matching** that handles merged SPF records (customers combining multiple provider `include:` directives in one TXT record)
- **78 tryout assertions** across two test files covering factory routing, per-provider record generation, DNS matching edge cases, operation orchestration with mock strategy, state persistence, and error paths

### What it does NOT include

This is the standalone operation layer only. No API routes (already in #2802), no mailer threading (#2804), no frontend (#2805).

### Reviewer notes

- SES DKIM selectors are placeholders (`ses-dkim-token-{1,2,3}`) — real values come from the AWS `VerifyDomainDkim` API response at runtime
- DNS timeouts are caught and treated as "record not found" (no retry logic; single-shot verification)
- Adding a new provider requires only a new strategy file + registration in `PROVIDER_MAP`

Closes #2803

## Test plan

- [x] `bundle exec try --agent try/unit/domain_validation/sender_strategies_try.rb` — 43 pass
- [x] `bundle exec try --agent try/unit/operations/validate_sender_domain_try.rb` — 35 pass
- [ ] Verify SES DKIM placeholder token format matches AWS API response shape
- [ ] Confirm SPF merge matching works with real multi-provider SPF records

### CODE REVIEW

File: lib/onetime/domain_validation/sender_strategies/strategy.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+          strategy_class = PROVIDER_MAP[normalized]
+
+          unless strategy_class
+            raise ArgumentError,
+              "Unknown mail provider: '#{provider_type}'. " \
+              "Valid options: #{PROVIDER_MAP.keys.join(', ')}"
+          end
+
+          strategy_class.new
```
[the remark this answers is not in the document]
Comment (delano) [reply lib/onetime/domain_validation/sender_strategies/strategy.rb:52]: Addressed the documentation side in 80536d218 — added a comment above `strategy_class.new` noting the no-arg construction contract.

The `**kwargs` forwarding is tracked in #2833 (exposing provider-specific params like SES region through the factory/operation interface).
